src/utils.ts

```diff
@@ -46,3 +48,92 @@ export function setContext(kubeconfig: string): string {
    kc.setCurrentContext(context)
    return kc.exportConfig()
 }
+
+/**
+ * Sets the context by pulling kubeconfig via az command
+ * @param admin Boolean for determining whether or not user is admin
+ * @param kubeconfigPath Path to place kubeconfig
+ * @returns Promise for the resulting exitCode number from running az command
+ */
+export async function azSetContext(
+   admin: boolean,
+   kubeconfigPath: string
+): Promise<number> {
+   const AZ_TOOL_NAME = 'az'
+   const AZ_USER_AGENT_ENV = 'AZURE_HTTP_USER_AGENT'
+   const AZ_USER_AGENT_ENV_PS = 'AZUREPS_HOST_ENVIRONMENT'
+   const originalAzUserAgent = process.env[AZ_USER_AGENT_ENV] || ''
+   const originalAzUserAgentPs = process.env[AZ_USER_AGENT_ENV_PS] || ''
+
+   try {
+      // set az user agent
+      core.exportVariable(AZ_USER_AGENT_ENV, getUserAgent(originalAzUserAgent))
+      core.exportVariable(
+         AZ_USER_AGENT_ENV_PS,
+         getUserAgent(originalAzUserAgentPs)
+      )
+
+      // get inputs
+      const resourceGroupName: string = core.getInput('resource-group', {
```

Comment:
can we pass the inputs in as parameters to keep the core.getInput calls at root level? that way it will be a pure function only relying on its params to determine behavior

Reply:
Good suggestion!